students/Nick_Lenssen/lesson05/mailroom_3.py

```diff
@@ -0,0 +1,89 @@
+from sys import exit
+import pathlib
+from operator import itemgetter
+
+def get_donation_amount():
+    while True:
+        amount = input("\nPlease enter in the amount you want to donate or c to go to original prompt: ")
+        try:
+            if amount == 'c':
+                main()
+                #donor_db.pop()
+            else:
+                amount = int(amount)
+            return amount
+        except ValueError:
+            print ('\nPlease enter an integer amount to donate')
+
+def create_report():
+    print("\n{:<18}{:<6}{:<20}{}{:<25}{}{:<15}".format(*('Donor Name','|','Total Given','|','Num Gifts','|','Average Gift')))
+    print ('-'*90)
+    #donor_db_c = sorted(donor_db, key=itemgetter(1))
+    for key, value in sorted(donor_db.items(), key=itemgetter(1), reverse = True):
+        print ("{:<20} {:>2} {:>12} {:>17}{:>17}{:>12}".format(*(key, '$', round(sum(donor_db[key]),2), 
+                        len(donor_db[key]), '$',round(sum(donor_db[key])/len(donor_db[key]),1))))
+
+def send_thanks():
+    while True:
+        full_name = input("\n\nPlease enter a name for a Thank You to go out or c to go to orginal prompt: ")
+        if full_name == 'list':
+            for key in donor_db:
+                print (key)
+        elif full_name == 'c':
+            main()
+        else:
```

Comment:
re your question about main flow.

you can call your function here at the top:
```
amount = get_donation_amount()
if not amount:
    return   # exit this flow, will automatically return to the main program flow
```
idea is that your function's don't know about how the main program flow is managed and by calling main() again, you are creating multiple sub loops